### PR TITLE
PLT-166:Added support for spot instance in modules & added validation …

### PR DESCRIPTION
**Pull request by SivaanandM**
…for same

### CODE REVIEW

File: examples/resources/spectrocloud_cluster_eks/resource.tf [hunk 1]
```diff
@@ -64,10 +64,12 @@ resource "spectrocloud_cluster_eks" "cluster" {
 
   machine_pool {
     name          = "worker-basic"
-    count         = 1
-    instance_type = "t3.large"
+    count         = 4
+    instance_type = "t3.xlarge"
+    capacity_type = "spot"
```
Comment (nikchern) [inline examples/resources/spectrocloud_cluster_eks/resource.tf:69]: we have a separate example for spot type, let's update that.

File: spectrocloud/resource_cluster_aws.go [hunk 2]
```diff
@@ -245,9 +246,10 @@ func resourceClusterAws() *schema.Resource {
 							Required: true,
 						},
 						"capacity_type": {
-							Type:     schema.TypeString,
-							Default:  "on-demand",
-							Optional: true,
+							Type:         schema.TypeString,
+							Default:      "on-demand",
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"on-demand", "spot"}, false),
```
Comment (nikchern) [inline spectrocloud/resource_cluster_aws.go:252]:
```suggestion
							ValidateFunc: validation.StringInSlice([]string{"on-demand", "spot"}, false),
							Description:  "Capacity type is an instance type,  can be 'on-demand' or 'spot'. Defaults to 'on-demand'.",

							
```

File: spectrocloud/resource_cluster_eks.go [hunk 3]
```diff
@@ -302,9 +302,10 @@ func resourceClusterEks() *schema.Resource {
 							Required: true,
 						},
 						"capacity_type": {
-							Type:     schema.TypeString,
-							Default:  "on-demand",
-							Optional: true,
+							Type:         schema.TypeString,
+							Default:      "on-demand",
+							Optional:     true,
+							ValidateFunc: validation.StringInSlice([]string{"on-demand", "spot"}, false),
```
Comment (nikchern) [inline spectrocloud/resource_cluster_eks.go:308]: here same as above. I suggest to actually move common parts of schema into a separate file. Could you please check how it's done in the example schemas/location_cluster_config.go ?

File: examples/resources/spectrocloud_cluster_eks/resource.tf (hunk 1 above)
Comment (nikchern) [reply examples/resources/spectrocloud_cluster_eks/resource.tf:69]: https://github.com/spectrocloud/terraform-provider-spectrocloud/blob/af95194e7b7e366159ef20b16fe9f45cdf347a08/examples/e2e/eks_spot

**Review by nikchern — CHANGES_REQUESTED** [review]

File: examples/resources/spectrocloud_cluster_eks/resource.tf (hunk 1 above)
Comment (SivaanandM) [reply examples/resources/spectrocloud_cluster_eks/resource.tf]: reverted to old example

**Review by nikchern — APPROVED** [review]

**Pull request merged**